From d3401371dc5a6eb029c97c5a75ed54c0bfb80780 Mon Sep 17 00:00:00 2001
From: Vijay kumar Tumati <vtumati@codeaurora.org>
Date: Fri, 17 Jan 2014 12:26:55 +0530
Subject: [PATCH 093/176] Camera: Send snapshot frame for reprocess if crop
 feature is enabled

If none of the other PP features are
enabled, Snapshot frames are not being
sent to reprocess and outpur jpeg
doesn't have zoom. This will fix that.

Change-Id: I7e4af3f18bb3164a03b2052f411161e645a5325b
---
 QCamera2/HAL/QCamera2HWI.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/QCamera2/HAL/QCamera2HWI.cpp b/QCamera2/HAL/QCamera2HWI.cpp
index 785f322..5ac642c 100644
--- a/QCamera2/HAL/QCamera2HWI.cpp
+++ b/QCamera2/HAL/QCamera2HWI.cpp
@@ -1,4 +1,4 @@
-/* Copyright (c) 2012-2013, The Linux Foundataion. All rights reserved.
+/* Copyright (c) 2012-2014, The Linux Foundataion. All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are
@@ -3801,6 +3801,10 @@ QCameraReprocessChannel *QCamera2HardwareInterface::addOnlineReprocChannel(
             pp_config.sharpness = mParameters.getInt(QCameraParameters::KEY_QC_SHARPNESS);
         }
 
+        if (gCamCapability[mCameraId]->min_required_pp_mask & CAM_QCOM_FEATURE_CROP) {
+            pp_config.feature_mask |= CAM_QCOM_FEATURE_CROP;
+        }
+
         if (mParameters.isWNREnabled()) {
             pp_config.feature_mask |= CAM_QCOM_FEATURE_DENOISE2D;
             pp_config.denoise2d.denoise_enable = 1;
-- 
1.8.3.1

